packages/web-components/fast-ssr/README.md

```diff
@@ -1,7 +1,150 @@
+const result = templateRenderer.render(html`
+    <!DOCTYPE HTML>
+    <html>
+        <body>
+            <my-element></my-element>
+        </body>
+    </html>
+`, defaultRenderInfo);
+```
+
+The template being rendered can also be a fragment of HTML, it does not need to be a valid document:
+
+```js
+const result = templateRenderer.render(html`<my-element></my-element>`, defaultRenderInfo);
+```
+
+#### Rendering Strings
+The template renderer can also render `string` types just as it would a template:
+
+```js
+const result = templateRenderer.render("<!DOCTYPE HTML><html><body><my-element></my-element></body></html>", defaultRenderInfo);
+```
+
+#### Rendering Templates with Bindings
+A template can be rendered with arbitrary source data by providing that source as the third argument to `.render()`. When done so, bindings are invoked with that source data:
+
+```ts
+const result = templateRenderer.render(html`
+    <h1>${x => x.message}</h1>
+`, defaultRenderInfo, { message: "hello world" });
+```
+
+#### Rendering Templates with Directives
+The `TemplateRenderer` can render templates with directives such as `when` and `repeat`. Some directives are no-ops, see the [directive support matrix](#directive-support-matrix) for details.
+
+```ts
+import { when, repeat } from "@microsoft/fast-element";
+
+const result = templateRenderer.render(html`
+    ${when(x => x.shouldRender, html`
+        <h1>Colors of a pixel</h1>
+        ${repeat(x => x.data, html`<li>${color => color}</li>`)}
+    `)}
+`, defaultRenderInfo, { shouldRender: true, data: ["red", "green", "blue"] });
+```
+
+##### Directive Support Matrix
+
+|Directive Name|Support|
+|-|-|
+|[`children`](https://www.fast.design/docs/fast-element/using-directives#the-children-directive)|[![](https://img.shields.io/badge/-Unsupported-red)]()|
+|[`ref`](https://www.fast.design/docs/fast-element/using-directives#the-ref-directive)|[![](https://img.shields.io/badge/-Unsupported-red)]()|
+|[`repeat`](https://www.fast.design/docs/fast-element/using-directives#the-repeat-directive)|[![](https://img.shields.io/badge/-Supported-brightgreen)]()|
+|[`slotted`](https://www.fast.design/docs/fast-element/using-directives#the-slotted-directive)|[![](https://img.shields.io/badge/-Unsupported-red)]()|
+|[`when`](https://www.fast.design/docs/fast-element/using-directives#the-when-directive)|[![](https://img.shields.io/badge/-Supported-brightgreen)]()|
+
+> Unsupported directives are no-ops. To understand more about why, see the [Design Philosophy.](#design-philosophy)
+
+### Configuring the RenderInfo Object
+`TemplateRenderer.render()` must be invoked with a `RenderInfo` object. It's purpose is to provide different element renderers to the process, as well as metadata about the rendering process. It can be used to render custom elements from different templating libraries in the same process. A pre-generated object is created for you by the factory function, but you can also easily construct your own: 
```

Comment:
```suggestion
`TemplateRenderer.render()` must be invoked with a `RenderInfo` object. Its purpose is to provide different element renderers to the process, as well as metadata about the rendering process. It can be used to render custom elements from different templating libraries in the same process. A pre-generated object is created for you by the factory function, but you can also easily construct your own: 
```